pkg/scheduler/factory_test.go

```diff
@@ -284,124 +304,146 @@ func TestDefaultErrorFunc(t *testing.T) {
+				queue.Add(testPod)
+				queue.Pop()
+			},
+			deferFn: func() { queue.Delete(testPod) },
+			expect:  testPod,
+		},
+		{
+			name: "pod is not updated during a scheduling cycle",
+			initFn: func() {
+				queue.Add(testPod)
+				queue.Pop()
+			},
+			deferFn:     func() { queue.Delete(testPod) },
+			expect:      testPod,
+			testBackoff: true,
+		},
+		{
+			name: "pod is deleted during a scheduling cycle",
+			initFn: func() {
+				queue.Add(testPod)
+				queue.Pop()
+			},
+			deferFn:                    func() { queue.Delete(testPod) },
+			podDeletedDuringScheduling: true,
+			expect:                     nil,
+		},
 	}
 
-	// Remove the pod from priority queue to test putting error
-	// pod in backoff queue.
-	queue.Delete(testPod)
-
-	// Trigger a move request
-	queue.MoveAllToActiveOrBackoffQueue("test")
-
-	// Trigger error handling again to put the pod in backoff queue
-	errFunc(testPodInfo, nil)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.initFn != nil {
+				tt.initFn()
+			}
+			if tt.deferFn != nil {
+				defer tt.deferFn()
+			}
 
-	foundPodFlag = false
-	for i := 0; i < maxIterations; i++ {
-		time.Sleep(100 * time.Millisecond)
-		// The pod should be found from backoff queue at this time
-		got := getPodfromPriorityQueue(queue, testPod)
-		if got == nil {
-			continue
-		}
+			// If pod is updated during the scheduling cycle.
+			if tt.podUpdatedDuringScheduling {
+				queue.Update(testPod, testPodUpdated)
+			} else if tt.podDeletedDuringScheduling {
+				// Simulate deleting Pod from API server.
+				client.CoreV1().Pods(testPod.Namespace).Delete(context.TODO(), testPod.Name, metav1.DeleteOptions{})
+				queue.Delete(testPod)
+			}
+			testPodInfo := &framework.QueuedPodInfo{Pod: testPod}
+			errFunc(testPodInfo, tt.injectErr)
+
+			if tt.expectNode != nil {
+				for _, n := range schedulerCache.Dump().Nodes {
+					if diff := cmp.Diff(tt.expectNode, n.Node()); diff != "" {
+						t.Errorf("Unexpected nodes (-want, +got): %s", diff)
+					}
+				}
+				return
+			}
 
-		testClientGetPodRequest(client, t, testPod.Namespace, testPod.Name)
+			var got *v1.Pod
+			if tt.podUpdatedDuringScheduling {
+				head, e := queue.Pop()
+				if e != nil {
+					t.Fatalf("Cannot pop pod from the activeQ: %v", e)
+				}
+				got = head.Pod
+			} else {
+				got = getPodFromPriorityQueue(queue, testPod)
+			}
 
-		if e, a := testPod, got; !reflect.DeepEqual(e, a) {
-			t.Errorf("Expected %v, got %v", e, a)
-		}
+			if diff := cmp.Diff(tt.expect, got); diff != "" {
+				t.Errorf("Unexpected pod (-want, +got): %s", diff)
+			}
 
-		foundPodFlag = true
-		break
-	}
+			if !tt.testBackoff {
+				return
+			}
 
-	if !foundPodFlag {
-		t.Errorf("Failed to get pod from the backoff queue after waiting for a minute: %v", testPod)
+			// Trigger a move request to move the failed pod from unschedulableQ to backoffQ.
+			queue.MoveAllToActiveOrBackoffQueue("test")
+			got = getPodFromPriorityQueue(queue, testPod)
```

Comment:
this looks for the pod in all queues so MoveAllToActiveOrBackoffQueue is not necessary and we are not really able to verify that the pod was in a specific queue.

Reply:
True, as this package cannot access the private backoffQ or unschedulableQ field.

I just remote this piece of code, any thoughts?